### ISSUE TITLE
Slicing in `torch.no_grad()`: erroneously requires gradient (without accumulator)

### ISSUE THREAD

**Issue by patrick-kidger** [pytorch/pytorch#45301]
## 🐛 Bug/Reproduce:

Slicing only sort-of-respects `no_grad()`. The resulting tensor does not have a history, but does require gradient. However using it does not then materialise a gradient accumulator.

```python
import torch
x = torch.rand(2, 2, requires_grad=True)
with torch.no_grad():
    y = x[:, :1]

z = y.clone()
print(y.requires_grad)  # True
print(z.grad_fn.next_functions)  # ((None, 0),)
```
For context, this came up during a custom `autograd.Function.forward` where there's an implicit `no_grad()`.

## Expected behavior

For `y` not to require gradient.

## Environment

 - PyTorch Version (e.g., 1.0): 1.6.0
 - OS (e.g., Linux): Linux
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Python version: 3.8.3

**Comment by zou3519** [comment]
I'm closing this issue as a duplicate. @patrick-kidger you should feel free to voice your support in the other thread (#11390)
